internal/mdns/dialable.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package mdns
+
+import (
+	"errors"
+	"fmt"
+	"net"
+
+	manet "github.com/multiformats/go-multiaddr/net"
+)
+
+var (
+	ErrTCPListenAddressNotFound = errors.New("TCP listen address not found")
+)
+
+func getMDNSIPsAndPort(p2pHost Networker) (ips []net.IP, port uint16) {
```

Comment:
nit:
```suggestion
func getMDNSIPsAndPort(networker Networker) (ips []net.IP, port uint16) {
```

Reply:
Changed to `network interfaceListenAddressesGetter` where `interfaceListenAddressesGetter` is an interface the `p2phost.Network()` returned value implements